src/external/fees/FeeManager.sol

```diff
@@ -0,0 +1,282 @@
+        }
+    }
+
+    //---------------------------
+    // Fees
+
+    /// @inheritdoc IFeeManager
+    function getDefaultCollateralFee() external view returns (uint) {
+        return defaultCollateralFee;
+    }
+
+    /// @inheritdoc IFeeManager
+    function getDefaultIssuanceFee() external view returns (uint) {
+        return defaultIssuanceFee;
+    }
+
+    /// @inheritdoc IFeeManager
+    function getCollateralWorkflowFee(
+        address workflow,
+        address module,
+        bytes4 functionSelector
+    ) public view returns (uint fee) {
+        bytes32 moduleFunctionHash =
+            getModuleFunctionHash(module, functionSelector);
+
+        //In case workflow fee is set return it
+        if (workflowCollateralFees[workflow][moduleFunctionHash].set) {
+            return workflowCollateralFees[workflow][moduleFunctionHash].value;
+        } //otherwise return default fee
+        else {
+            return defaultCollateralFee;
+        }
+    }
+
+    /// @inheritdoc IFeeManager
+    function getIssuanceWorkflowFee(
+        address workflow,
+        address module,
+        bytes4 functionSelector
+    ) public view returns (uint fee) {
+        bytes32 moduleFunctionHash =
+            getModuleFunctionHash(module, functionSelector);
+
+        //In case workflow fee is set return it
+        if (workflowIssuanceFees[workflow][moduleFunctionHash].set) {
+            return workflowIssuanceFees[workflow][moduleFunctionHash].value;
+        } //otherwise return default fee
+        else {
+            return defaultIssuanceFee;
+        }
+    }
+
+    /// @inheritdoc IFeeManager
+    function getCollateralWorkflowFeeAndTreasury(
+        address workflow,
+        address module,
+        bytes4 functionSelector
+    ) external view returns (uint fee, address treasury) {
+        return (
+            getCollateralWorkflowFee(workflow, module, functionSelector),
+            getWorkflowTreasuries(workflow)
+        );
+    }
+
+    /// @inheritdoc IFeeManager
+    function getIssuanceWorkflowFeeAndTreasury(
+        address workflow,
+        address module,
+        bytes4 functionSelector
+    ) external view returns (uint fee, address treasury) {
+        return (
+            getIssuanceWorkflowFee(workflow, module, functionSelector),
+            getWorkflowTreasuries(workflow)
+        );
+    }
+
+    //--------------------------------------------------------------------------
+    // Setter Functions
+
+    //---------------------------
+    // Treasuries
+
+    /// @inheritdoc IFeeManager
+    function setDefaultProtocolTreasury(address _defaultProtocolTreasury)
+        external
+        onlyOwner
+        validAddress(_defaultProtocolTreasury)
+    {
+        defaultProtocolTreasury = _defaultProtocolTreasury;
+        emit DefaultProtocolTreasurySet(_defaultProtocolTreasury);
+    }
+
+    /// @inheritdoc IFeeManager
+    function setWorkflowTreasuries(address workflow, address treasury)
```

Comment:
As this just sets the on treasury of one workflow, should it be called `setWorkflowTreasury()` instead of `setWorkflowTreasuries()`?